lib/template-parts.php

```diff
@@ -191,15 +182,52 @@ function filter_rest_wp_template_part_collection_params( $query_params ) {
  */
 function filter_rest_wp_template_part_query( $args, $request ) {
 	if ( $request['theme'] ) {
-		$meta_query   = isset( $args['meta_query'] ) ? $args['meta_query'] : array();
-		$meta_query[] = array(
-			'key'   => 'theme',
-			'value' => $request['theme'],
+		$tax_query   = isset( $args['tax_query'] ) ? $args['tax_query'] : array();
+		$tax_query[] = array(
+			'taxonomy' => 'wp_theme',
+			'field'    => 'slug',
+			'terms'    => $request['theme'],
 		);
 
-		$args['meta_query'] = $meta_query;
+		$args['tax_query'] = $tax_query;
```

Comment:
I don't think we need this filter anymore, since we can just use the built-in `wp_theme` param now.

I'm wondering about this.
Now we should be able to filter by taxonomy by simply providing `?wp_theme=[ termId ]`.
Of course we don't have the term ID, but its slug.
Maybe there's a hidden trick that lets us do this without having to filter the query? 🤔

Reply:
:laughing: I ran into the same issue in my alternative PR a few months ago and after some research I couldn't find a better way to do it either... (https://github.com/WordPress/gutenberg/pull/24720#issuecomment-702675209)
We _could_ do it by adding an extra JS implementation to get the term ID from its slug, but that would be extremely wasteful as it would just result in an extra query to the server... The current implementation is not perfect, but I think it's the best we can do with the current state of the REST API without significant refactors :+1:

Reply:
Ooh, I see, makes sense.